axiom/nr_segment_datastore.c

```diff
@@ -12,6 +12,36 @@
 #include "util_sql.h"
 #include "util_logging.h"
 
+static bool create_instance_metric(nr_segment_t* segment,
+                                   const char* product,
+                                   nr_segment_datastore_t* datastore,
+                                   const nr_datastore_instance_t* instance) {
+  /*
+   * If a datastore instance was provided, we need to add the relevant data to
+   * the segment and the relavant metrics.
+   */
+  nrtxn_t* txn = segment->txn;
```

Comment:
Check for `segment` being `NULL`?

Reply:
the segment is checked for NULL by the caller

Reply:
We don't have a good convention currently to alert when we can depend on a function to accept `NULL` or not.  In this case it is a local static function so perhaps it is tightly coupled to the calling code in the same file and the `NULL` check could be relaxed.  I am just concerned in general that it is possible for new code to call these functions and cause a segfault if they also do not enforce the NULL check.  I've always leaned towards robustness at all levels versus the very slight overhead of redundant checking.  Not sure how to resolve this for this PR or the agent code in general.